src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -658,9 +658,18 @@ private void doHandleAnonymous(HttpServletRequest request,
             } else {
                 String containerName = path[1];
                 String blobName = path[2];
-                BlobAccess access = blobStore.getBlobAccess(containerName,
-                        blobName);
-                if (access != BlobAccess.PUBLIC_READ) {
+                boolean publicAccess = false;
+                String blobStoreType = getBlobStoreType(blobStore);
+                if (Quirks.NO_BLOB_ACCESS_CONTROL.contains(blobStoreType)) {
+                    ContainerAccess access = blobStore.getContainerAccess(
+                            containerName);
+                    publicAccess = access == ContainerAccess.PUBLIC_READ;
+                } else {
+                    BlobAccess access = blobStore.getBlobAccess(containerName,
+                            blobName);
+                    publicAccess = access == BlobAccess.PUBLIC_READ;
+                }
+                if (!publicAccess) {
```

Comment:
Replicate this logic for HEAD object below.  Factor out the check into a function named `checkPublicAccess`.